### PR TITLE
chore: added a functional test for the Kotlin SDK

### PR DESCRIPTION
**Pull request by jkaster**
Tests creating user attributes and user attribute group values.

And deleting them.

### CODE REVIEW

**Review by bryans99 — COMMENTED** [review]
I'm confused by this!

File: kotlin/src/test/TestMethods.kt [hunk 1]
```diff
@@ -549,12 +549,22 @@ class TestMethods {
...
+            val attrGroup = UserAttributeGroupValue(group_id=group.id!!, user_attribute_id=attribute.id!!, value_is_hidden=false, value="1.2.3.4")
+            sdk.ok<Array<UserAttributeGroupValue>>(sdk.set_user_attribute_group_values(attribute.id!!, arrayOf(attrGroup))).first()
             // We won't get here when there's an error
-            sdk.ok(sdk.delete_user_attribute(actual.id!!))
+            sdk.delete_user_attribute_group_value(group.id!!, attribute.id!!)
+            sdk.delete_user_attribute(attribute.id!!)
         } catch (e: java.lang.Error) {
             val msg = e.toString()
-            assertTrue(msg.contains("POST /user_attributes"))
+            assertTrue(msg.contains("POST /user_attributes"), msg)
```
[the remark this answers is not in the document]
Comment (jkaster) [reply kotlin/src/test/TestMethods.kt:567]: if we *do* get a message and it's not the POST issue, this outputs what the actual message was.

File: kotlin/src/test/TestMethods.kt [hunk 2]
```diff
@@ -549,12 +549,22 @@ class TestMethods {
...
+            val groups = sdk.ok<Array<Group>>(sdk.all_groups("id, name"))
+            val group = groups.first()
+
+            val all = sdk.ok<Array<UserAttribute>>(sdk.all_user_attributes("id, name"))
+            var attribute = all.find { a -> a.name == body.name}
+            if (attribute == null) {
+                attribute = sdk.ok<UserAttribute>(sdk.create_user_attribute(body))
+            }
+            val attrGroup = UserAttributeGroupValue(group_id=group.id!!, user_attribute_id=attribute.id!!, value_is_hidden=false, value="1.2.3.4")
+            sdk.ok<Array<UserAttributeGroupValue>>(sdk.set_user_attribute_group_values(attribute.id!!, arrayOf(attrGroup))).first()
```
Comment (bryans99) [inline kotlin/src/test/TestMethods.kt:561]: Are we expecting an error? If we are, should we throw an error here

File: kotlin/src/test/TestMethods.kt [hunk 3]
```diff
@@ -549,12 +549,22 @@ class TestMethods {
...
             // We won't get here when there's an error
-            sdk.ok(sdk.delete_user_attribute(actual.id!!))
+            sdk.delete_user_attribute_group_value(group.id!!, attribute.id!!)
```
Comment (bryans99) [inline kotlin/src/test/TestMethods.kt:563]: Do we need to assert these work?

Comment (jkaster) [reply kotlin/src/test/TestMethods.kt:563]: If they don't, the error will be output in the catch block

**Comment by jkaster** [comment]
This test is not needed

**Pull request closed without merging**